lib/montage/query.rb

```diff
@@ -8,9 +8,33 @@ class Query
     include Montage::Support
 
     attr_accessor :query
+    attr_reader :schema
 
-    def initialize
-      @query = { filter: {} }
+    # Initializes the query instance
+    #
+    # * *Attributes* :
+    #   - +schema+ -> The name of the schema you wish to query.  Alphanumeric
+    #     characters and underscores are allowed.
+    #   - +query+ -> A query hash containing desired options
+    # * *Returns* :
+    #   - A valid Montage::Query instance
+    # * *Raises* :
+    #   - +InvalidAttributeFormat+ -> If the declared schema is not a string or
+    #     contains non-alphanumeric characters.  Underscore ("_") is allowed!
+    #
+    def initialize(params = {})
+      @schema = params[:schema]
+      @query = { "$schema" => @schema, "$query" => { "$filter" => {} } }
+      fail InvalidAttributeFormat, "Schema attribute must be declared and valid!" unless schema_valid?
```

Comment:
I'm a big fan of not using parens in some cases, but I think this line of code would be more readable with the parens.